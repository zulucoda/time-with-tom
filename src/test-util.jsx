import React from 'react';
import { oneOfType, shape, element, func } from 'prop-types';
import { Form, Formik } from 'formik';
import { BrowserRouter as Router } from 'react-router-dom';

/**
 * Formik Wrapper Test Util
 * @param {React.FunctionComponent} Component
 * @param {Object} props
 * @param {Object} initialValues
 * @param {Function} onSubmit
 * @returns {Formik}
 */
export const FormikWrapperTestUtil = ({
  Component,
  props = {},
  initialValues = {},
  onSubmit = jest.fn(),
}) => (
  <Formik initialValues={initialValues} onSubmit={onSubmit}>
    <Form>
      <Component {...props} />
    </Form>
  </Formik>
);

FormikWrapperTestUtil.propTypes = {
  Component: oneOfType([element.isRequired, func]),
  props: shape({}),
  initialValues: shape({}),
  onSubmit: func,
};

/**
 * React Router Wrapper Test Util
 * @param {React.FunctionComponent} Component
 * @param {object} props
 * @returns {React.FunctionComponent}
 */
export const ReactRouterWrapperTestUtil = ({ Component, props = {} }) => (
  <Router>
    <Component {...props} />
  </Router>
);

ReactRouterWrapperTestUtil.propTypes = {
  Component: oneOfType([element.isRequired, func]),
  props: shape({}),
};
