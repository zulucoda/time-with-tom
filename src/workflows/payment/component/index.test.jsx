import React from 'react';
import { render } from '@testing-library/react';
import Payment from '.';
import { FormikWrapperTestUtil } from '../../../test-util';

describe('Workflow - Payment - Component - Activity - Unit Test', function () {
  test('does it render', () => {
    const { container } = render(<FormikWrapperTestUtil Component={Payment} />);
    expect(container).toMatchSnapshot();
  });
});
