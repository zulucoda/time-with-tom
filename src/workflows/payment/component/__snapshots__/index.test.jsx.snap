// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Workflow - Payment - Component - Activity - Unit Test does it render 1`] = `
<div>
  <form
    action="#"
  >
    <div
      class="sc-AxirZ dyMoOF"
    >
      <div
        class="sc-AxjAm jkaSBu"
      >
        <div
          class="MuiFormControl-root MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink="false"
          >
            Cardholder Name
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input"
              name="ccName"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="sc-AxjAm jkaSBu"
      >
        <div
          class="MuiFormControl-root MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink="false"
          >
            Card Number
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input"
              name="ccNumber"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="sc-AxiKw fKluip"
      >
        <div
          class="sc-AxjAm jkaSBu"
        >
          <div
            class="MuiFormControl-root MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink="false"
            >
              Expiry Month
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input"
                name="ccExpiryMonth"
                type="number"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="sc-AxjAm jkaSBu"
        >
          <div
            class="MuiFormControl-root MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink="false"
            >
              Expiry Year
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input"
                name="ccExpiryYear"
                type="number"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="sc-AxjAm jkaSBu"
        >
          <div
            class="MuiFormControl-root MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink="false"
            >
              CVC
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input"
                name="ccCvc"
                type="number"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;
